tipg/settings.py

```diff
@@ -131,18 +131,20 @@ class PostgresSettings(BaseSettings):
 
     # https://github.com/tiangolo/full-stack-fastapi-postgresql/blob/master/%7B%7Bcookiecutter.project_slug%7D%7D/backend/app/app/core/config.py#L42
     @field_validator("database_url", mode="before")
-    def assemble_db_connection(cls, v: Optional[str], info: FieldValidationInfo) -> Any:
+    def assemble_db_connection(
+        cls, v: Optional[str], info: FieldValidationInfo
+    ) -> PostgresDsn:
         """Validate db url settings."""
         if isinstance(v, str):
-            return v
+            return PostgresDsn(v)
 
         return PostgresDsn.build(
             scheme="postgresql",
             username=info.data.get("postgres_user"),
             password=info.data.get("postgres_pass"),
             host=info.data.get("postgres_host", ""),
             port=info.data.get("postgres_port", 5432),
```

Comment:
pydantic will always cast the host to `integer` so there is no real need to add `int()` here, even if it would be technically correct.